src/main/java/io/github/oliviercailloux/y2018/apartments/valuefunction/profile/Profile.java

```diff
@@ -0,0 +1,387 @@
+  private Range<Double> floorAreaTerraceWeightRange;
+  private Range<Double> wifiSubjectiveValueWeightRange;
+  private Range<Double> pricePerNightSubjectiveValueWeightRange;
+  private Range<Double> nbMinNightSubjectiveValueWeightRange;
+  private Range<Double> teleSubjectiveValueWeightRange;
+
+  private Profile() {
+    this.linearAvf = null;
+    this.floorAreaWeightRange = null;
+    this.nbBedroomsWeightRange = null;
+    this.nbSleepingWeightRange = null;
+    this.nbBathroomsWeightRange = null;
+    this.terraceWeightRange = null;
+    this.floorAreaTerraceWeightRange = null;
+    this.wifiSubjectiveValueWeightRange = null;
+    this.pricePerNightSubjectiveValueWeightRange = null;
+    this.nbMinNightSubjectiveValueWeightRange = null;
+    this.teleSubjectiveValueWeightRange = null;
+  }
+
+  /**
+   * This function allows the user to clone an object Profile
+   *
+   * @return an object Profile
+   */
+  private Profile cloneProfile() {
+
+    Profile prof = new Profile();
+
+    prof.linearAvf = this.linearAvf;
+    prof.floorAreaWeightRange = this.floorAreaWeightRange;
+    prof.nbBedroomsWeightRange = this.nbBedroomsWeightRange;
+    prof.nbSleepingWeightRange = this.nbSleepingWeightRange;
+    prof.nbBathroomsWeightRange = this.nbBathroomsWeightRange;
+    prof.terraceWeightRange = this.terraceWeightRange;
+    prof.floorAreaTerraceWeightRange = this.floorAreaTerraceWeightRange;
+    prof.wifiSubjectiveValueWeightRange = this.wifiSubjectiveValueWeightRange;
+    prof.pricePerNightSubjectiveValueWeightRange = this.pricePerNightSubjectiveValueWeightRange;
+    prof.nbMinNightSubjectiveValueWeightRange = this.nbMinNightSubjectiveValueWeightRange;
+    prof.teleSubjectiveValueWeightRange = this.teleSubjectiveValueWeightRange;
+
+    return prof;
+  }
+
+  /**
+   * Gives the linearAVF of a Profile
+   *
+   * @return the LinearAVF of the Profile
+   */
+  public LinearAVF getLinearAVF() {
+    return this.linearAvf;
+  }
+
+  /**
+   * Gives the subjective value weight of a criterion
+   *
+   * @param crit the criterion we want to know the value
+   * @return the subjective value weight
+   */
+  public Range<Double> getWeightRange(Criterion crit) {
+    switch (crit) {
+      case TELE:
+        return teleSubjectiveValueWeightRange;
+      case TERRACE:
+        return terraceWeightRange;
+      case WIFI:
+        return wifiSubjectiveValueWeightRange;
+      case FLOOR_AREA:
+        return floorAreaWeightRange;
+      case FLOOR_AREA_TERRACE:
+        return floorAreaTerraceWeightRange;
+      case NB_BATHROOMS:
+        return nbBathroomsWeightRange;
+      case NB_BEDROOMS:
+        return nbBedroomsWeightRange;
+      case NB_SLEEPING:
+        return nbSleepingWeightRange;
+      case NB_MIN_NIGHT:
+        return nbMinNightSubjectiveValueWeightRange;
+      case PRICE_PER_NIGHT:
+        return pricePerNightSubjectiveValueWeightRange;
+      default:
+        throw new IllegalArgumentException();
```

Comment:
`VerifyException`conviendrait mieux : on est en droit de supposer que toutes les valeurs possibles de `Crit` sont fixées une fois pour toutes et ne changent pas. OPT.

Reply:
Changed in commit a4ff859